src/models.rs

```diff
@@ -24,10 +24,32 @@ use serde::{Deserialize, Serialize};
 use crate::{clients::detector::ContextType, pb};
 
 /// Parameters relevant to each detector
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
-pub struct DetectorParams {
-    /// Threshold with which to filter detector results by score
-    pub threshold: Option<f64>,
+#[derive(Clone, Debug, Default, PartialEq, Serialize, Deserialize)]
+pub struct DetectorParams(HashMap<String, serde_json::Value>);
+
+impl DetectorParams {
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+
+    pub fn threshold(&self) -> Option<f64> {
+        self.0.get("threshold").and_then(|v| v.as_f64())
```

Comment:
what sort of error we would get if `threshold` is provided in the request but a different data type? like if someone does `threshold: "0.0"` ?

Reply:
It would return `None` as it would be `Value::String` instead of `Value::Number`. `Value::as_f64()` will only return `f64` for numbers. Clients should be encouraged to use the proper data type.

Reply:
Just added detector params validation, including a rule for this.